addons/mrp/models/mrp_workorder.py

```diff
@@ -895,3 +928,18 @@ def _compute_expected_operation_cost(self):
 
     def _compute_current_operation_cost(self):
         return (self.get_duration() / 60.0) * (self.costs_hour or self.workcenter_id.costs_hour)
+
+    def _are_previous_workorder_blocking(self):
+        return any(w.state not in ('done', 'cancel') for w in self.blocked_by_workorder_ids)
+
+    def _has_qty_ready(self):
+        return float_compare(self.qty_ready, 0, precision_rounding=self.product_uom_id.rounding) > 0
+
+    def _is_partially_produced(self, strict=False):
+        partially_produced = float_compare(self.qty_produced, 0, precision_rounding=self.product_uom_id.rounding) > 0
+        if strict:
+            partially_produced &= float_compare(self.qty_produced, self.qty_production, precision_rounding=self.product_uom_id.rounding) < 0
+        return partially_produced
+
+    def _has_started(self):
+        return self.date_start and self.time_ids
```

Comment:
State transitions of the WO have changed and so, we can now go back and forth to the 'progress' state, we need this method to rewrite the state as 'progress' instead of 'ready' when such transition happens.